A crash while entering text in the Editor

### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

From Discord:
I got a crash in editor when I tried to type something in map description

`22.05.2024 22:50:47: [ERROR]    main:  Exception 'basic_string::insert: __pos (which is 1) > this->size() (which is 0)' occurred during application runtime.`

### Save file

No need.

### Additional info

_No response_

--- a/src/fheroes2/gui/ui_text.cpp
+++ b/src/fheroes2/gui/ui_text.cpp
@@ -945,7 +945,8 @@ namespace fheroes2
 
         if ( pointerLine >= static_cast<int32_t>( lineInfos.size() ) ) {
             // Pointer is lower than the last text line.
-            return textSize;
+            // Reduce textSize by 1 because the cursor character ('_') was added to the line.
+            return textSize - 1;
         }
 
         size_t cursorPosition = 0;
@@ -980,7 +981,8 @@ namespace fheroes2
             }
         }
 
-        return textSize;
+        // Reduce textSize by 1 because the cursor character ('_') was added to the line.
+        return textSize - 1;
     }
 
     bool isFontAvailable( const std::string & text, const FontType fontType )

